02-intermediate-ruby/api-intro.md

```diff
@@ -1,7 +1,215 @@
+
+However, since the API doesn't send any information about how to display the data, the browser just displays it in a raw form. This can be hard to read for us humans.
+
+To help us out, we will install the [JSON Viewer chrome extension](https://chrome.google.com/webstore/detail/json-viewer/gbmdgpbipfallnflgajpaliibnhdgobh). This handy tool will format JSON in an easy-to-read way. Try it out now!
+
+## The Request/Response Cycle
+
+<!-- NB: if you update this text, please also update the similar text in ../08-rails/requests-and-responses.md -->
+
+One of the most important ideas in web development is the _request-response cycle_:
+
+1. **User** (a human) takes some action (run a command-line program) to begin the cycle
+1. **Client** (our program) makes a _request_ to the server
+1. **Server** (a computer on the internet) receives the request and processes it
+1. **Server** sends a _response_ for the request to the client
+1. **Client** receives the response and displays it to the user (printing text to the command line)
+1. **User** takes another action, and the cycle begins anew
+
+We can shorten this to 3 major steps:
+1. Request
+1. Processing
+1. Response
+
+![Request response overview](images/apis/request_response_overview.png)
+<!-- https://www.draw.io/#G17ynKwAoksm5XSU5tqEsDiGuMz_UwZByl -->
+
+We won't worry too much about the processing step yet - that will be the subject of our Rails unit. For now we will focus on the request and the response.
+
+## Request
+
+The first part of the request-response cycle is the request. A request is made by the browser to the Rails server. Here is an example of a real-life request:
+
+```
+GET /books HTTP/1.1
+User-Agent: curl/7.30.0
+Host: localhost:3000
+Content-Length: 6
+Accept: */*
+```
+
+This is the literal text that is sent over the internet by the browser. There are two components of the request that are important to us right now:
+- `GET`, an HTTP verb
+- `/books`, path
+
+There's a bunch of other stuff in there too, but we won't be worrying about that for a while (if at all).
+
+### Verbs and Paths
+
+The _verb_ tells the server what sort of thing the client wants to do. There are a bunch of verbs, but the most common are:
+
+Verb             | Meaning
+---              | ---
+`GET`            | Send me some data
+`POST`           | Here is some data, please save it
+`PATCH` or `PUT` | Make a change to existing data
+`DELETE`         | Remove data
+
+There's quite a few default _verbs_ in the HTTP spec, though we (Ruby/Rails devs) mostly just use `GET`, `POST`, `PUT` (and `PATCH`), and `DELETE`.
+
+The _path_ tells the server what resource the client is interested in. In our example above this is the list of books at `/books`. Later we'll see it looking at individual books via the path `/books/3` (where 3 is the ID of the book in question).
```

Comment:
Since this is their first exposure to the idea of a "path," it might be interesting to give a small example before `/books` to relate the idea of a URL to a `path`